app/gui2/src/util/ast/extended.ts

```diff
@@ -51,16 +71,22 @@ export class AstExtended<T extends Tree | Token = Tree | Token, HasIdMap extends
     return Token.isInstance(this.inner) ? Token.typeNames[this.inner.type] : null
   }
 
-  isToken<T extends Ast.Token.Type>(
+  isToken<T extends OneOrArray<Ast.Token.Type>>(
```

Comment:
(as mentioned elsewhere) consider `<T>` `type?: OneOrArray<T>`

Reply:
It is important to have a `T` named here, as it is also used in returned value type. Writing it like this seems to be important to get correct type inference.

Reply:
right - i was hoping that changing the generic like this, would let us avoid `ExtractType` entirely